### PR TITLE
Create storage wrapper methods and added authentication persistance

### PR DESCRIPTION
**Pull request by ShinUrech**
Title: Refactor Authentication, Add Storage Tests, and Update Firebase Configuration

Description:

This PR includes several important updates and additions to our project:

1. Authentication Refactoring:
   - Updated import paths in auth tests
   - Consolidated mocks for better test isolation

2. Storage Functionality:
   - Added comprehensive tests for Firebase storage functions (upload/delete for images and videos)
   - Implemented error handling and edge cases in storage tests

3. Firebase Configuration:
   - Updated Firebase initialization to use React Native persistence
   - Added AsyncStorage for better data persistence in the app

4. Type Definitions:
   - Added TTenantData and TLandlordData type definitions
   - Introduced UserType enum for better type safety

5. Dependencies:
   - Updated react-native-safe-area-context and react-native-screens to specific versions for compatibility

6. Project Configuration:
   - Added Jest configuration for TypeScript support
   - Updated tsconfig.json to resolve Firebase auth types correctly

These changes improve our testing infrastructure, enhance type safety, and update our Firebase integration for better performance in a React Native environment.

Please review the changes, particularly the new storage tests and Firebase configuration updates. Let me know if any further modifications or explanations are needed.

### CODE REVIEW

File: types/types.tsx [hunk 1]
```diff
@@ -6,5 +6,11 @@ export type RootStackParamList = {
...
   
-declare type TTenantData ={}
-declare type TLandlordData ={}
+export declare type TTenantData ={}
+export declare type TLandlordData ={}
+
+export enum UserType {
+    TENANT,
+    LANDLORD,
+    UNAUTHENTICATED
```
Comment (mahdiatallah) [inline types/types.tsx:15]: I'm not sure I fully understand the purpose of the UNAUTHENTICATED user type since a user will always choose to either be a tenant or a landlord/manager at each sign-up.

Comment (ShinUrech) [reply types/types.tsx:15]: I added it to allow the handling of tri state authentications (since we have two different types of accounts, it allows to differentiate an unauthenticated user from a user connected with a TENANT account from a user with a LANDLORD account). This will come in handy when handling the authentication state changes.

Comment (mahdiatallah) [reply types/types.tsx:15]: Okay I see. Thanks for explaining it.

**Review by mahdiatallah — APPROVED** [review]
